cpp/src/c_api/graph_sg.cpp

```diff
@@ -207,6 +210,22 @@ struct create_graph_functor : public cugraph::c_api::abstract_functor {
                                         : false);
       }
 
+      if (symmetrize_) {
```

Comment:
In addition to this logic, if `symmetrize_` is true then we should override the graph properties to set `is_symmetric` to true.